lib/src/iterator_extension.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:flutter/widgets.dart';
+
+/// Extension on [Iterable] to add a separator between each widget using generators.
+extension IterableExt on Iterable<Widget> {
+  Iterable<Widget> separator(Widget element) sync* {
```

Comment:
On the `separator` you should add the doc for separator.

Reply:
Also, in the doc itself, add an example. For reference, you can check out the snackbar documentation